python/pyspark/errors/exceptions/base.py

```diff
@@ -31,6 +31,7 @@ def __init__(
         message: Optional[str] = None,
         error_class: Optional[str] = None,
         message_parameters: Optional[Dict[str, str]] = None,
+        data: Optional[Any] = None,
```

Comment:
Can we type? e.g., tuple and/or dict

Reply:
Sure. I initially set it as `Any` to allow taking other data types in other use cases, but currently we only take list of Rows.